.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,16 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/docker-existing-dockerfile
+{
+    "name": "Existing Dockerfile",
+    "build": {
+        "context": "..",
+        "dockerfile": "../Dockerfile"
+    },
+    "features": {
+        "ghcr.io/devcontainers/features/common-utils:latest": {},
+		"ghcr.io/devcontainers/features/docker-in-docker:latest": {},
+		"ghcr.io/devcontainers/features/rust:latest": {},
+        "ghcr.io/devcontainers-contrib/features/protoc:latest": {} 
+    },
+    "postCreateCommand": "apt-get update && apt-get install -y clang lld cmake protobuf-compiler jq && rustup component add rustfmt"
+}
```

Comment:
```suggestion
{
  "name": "Debian",
  "image": "mcr.microsoft.com/devcontainers/base:bullseye",

  "features": {
    "ghcr.io/devcontainers/features/common-utils:latest": {},
    "ghcr.io/devcontainers/features/docker-in-docker:latest": {},
    "ghcr.io/devcontainers/features/rust:latest": {
      "profile": "default"
    },
    "ghcr.io/devcontainers-contrib/features/protoc:latest": {}
  },
  "postCreateCommand": {
    "apt-get": "sudo apt-get update -y && sudo apt-get upgrade -y && sudo apt-get install -y clang lld cmake protobuf-compiler jq gcc g++",
    // mostly for running cargo +nightly fmt
    "nightly": "rustup toolchain install nightly"
  },

  // Use 'forwardPorts' to make a list of ports inside the container available locally.
  "forwardPorts": [6333, 6334]
}

```

Reply:
> I think we can avoid using and modifying the existing Dockerfile, as it is intended only for compiling production releases, which might explain why it takes so long to build a debug profile.

Yes, definitely, I was just about to make a commit with a separate Dockerfile:

```Dockerfile

ARG RUST_VERSION

FROM rust:${RUST_VERSION:-1}-bullseye

RUN apt-get update && \
    apt-get install -y ca-certificates tzdata \
    clang lld cmake protobuf-compiler jq 

RUN rustup toolchain install nightly && \
    rustup component add rustfmt --toolchain nightly

ARG APP=/qdrant

RUN mkdir -p ${APP}

WORKDIR ${APP}

EXPOSE 6333
EXPOSE 6334

CMD ["./entrypoint.sh"]
```

I think the resulting differences are marginal. Going with the JSON is probably easier to grasp.